File: Napakalaki Java/src/NapakalakiGame/Monster.java
package NapakalakiGame;

/**
 *
 * @author Jesús Sánchez de Lechina Tejada & Miguel Ángel Robles Urquiza
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badconsequence;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster(String name, int level, BadConsequence bc, Prize prize, int lvl_change){
        this.name = name;
        this.combatLevel = level;
        this.badconsequence = bc;
        this.prize = prize;
        this.levelChangeAgainstCultistPlayer = lvl_change;
    }
    
    public String getName(){ return name; }    
    public int getCombatLevel(){ return combatLevel; }    
    public int getCombatLevelAgainstCultistPlayer(){ return combatLevel + levelChangeAgainstCultistPlayer;}
    public BadConsequence getBadConsequence(){ return badconsequence; }
    public Prize getPrize(){ return prize; }  

    public int getLevelsGained(){ return prize.getLevel(); }    
    public int getTreasuresGained(){ return prize.getTreasures(); }    

    
    @Override
    public String toString(){
        return "Monster: " + name + " \nCombat level: " + Integer.toString(combatLevel) + " \nPrize: " + prize.toString() + "\nBad Consequence: " + badconsequence.toString() + "\n";
    }
}
    

File: Napakalaki Ruby/lib/monster.rb
# By: Miguel Robles Urquiza
#     Jesús Sánchez de Lechina Tejada

module NapakalakiGame
  class Monster

    def initialize(n, l, b, p, cl)
      @name = n
      @combatLevel = l
      @badConsequence = b
      @prize = p
      @levelChangeAgainstCultistPlayer = cl
    end   
    
    def getName
      @name
    end

    def getCombatLevel
      @combatLevel
    end
    
    def getBadConsequence         
      @badConsequence
    end
    
    def getTreasuresGained          # Número de treasures que se obtienen al ganar
      @prize.getTreasures.size()
    end

    def getLevelsGained             # Levels que se obtienen al ganar
      @prize.getLevels
    end
    
    def getLevelsGained         
      @prize.getLevels
    end
    
    def getTreasuresGained         
      @prize.getTreasures
    end
    
    def getCombatLevelAgainstCultistPlayer
      getCombatLevel + @levelChangeAgainstCultistPlayer
    end
    
    def to_s
      "Name: #{@name} \nCombatLevel: #{@combatLevel} (#{getCombatLevelAgainstCultistPlayer} si eres sectario) \nPrize: #{@prize} \nBadConsequence: #{@badConsequence} \n\n"
    end

  end
end